corehq/apps/userreports/views.py

```diff
@@ -446,21 +470,24 @@ def export_data_source(request, domain, config_id):
     config = get_document_or_404(DataSourceConfiguration, domain, config_id)
     table = get_indicator_table(config)
     q = Session.query(table)
-    column_headers = [col['name'] for col in q.column_descriptions]
+    columns = q.column_descriptions
+
+    try:
+        keyword_filters, sql_filters = process_url_params(request.GET, columns)
+    except BadSpecError as e:
+        return HttpResponse(e.message, status=400)
 
-    # apply filtering if any
-    filter_values = {key: value for key, value in request.GET.items() if key != 'format'}
-    for key in filter_values:
-        if key not in column_headers:
-            return HttpResponse('Invalid filter parameter: {}'.format(key), status=400)
-    q = q.filter_by(**filter_values)
+    q = q.filter_by(**keyword_filters)
+    for sql_filter in sql_filters:
+        q = q.filter(sql_filter)
 
     # build export
     def get_table(q):
-        yield column_headers
+        yield [col['name'] for col in columns]
         for row in q:
             yield row
 
+    return HttpResponse('Woo! {} rows'.format(q.count()))
```

Comment:
No, this is just for testing (this isn't ready for merge) - I didn't want to actually download the csv every time.  I've also hardcoded a date range above